src/misc/debug.c

```diff
@@ -141,7 +141,54 @@ void debug_vprintf_trace(const char* func_name,
 			 va_list ap)
 
 {
+#ifdef USE_LOG_BACKEND
+	char fmt_copy[1024] = "";
+	strcpy(fmt_copy, fmt);
+	// take care of the trailing newline often present...
+	if (fmt_copy[strlen(fmt)-1] == '\n') {
+		fmt_copy[strlen(fmt)-1] = '\0';
+	}
+
+	char tmp[1024] = {""};
+	vsnprintf(tmp, 1024, fmt, ap);
+
+	switch(level) {
+	case DP_ERR:
+#  ifdef USE_LOG_SIEMENS_BACKEND
```

Comment:
ok, how about a single 'vendor_log' function which takes a 'level' argument? And all different backends could simply export the same function (as they are mutually exclusive anyway). This would remove this boilerplate code.

Reply:
Sounds good, I will make the necessary changes.